CONTRIBUTING.md

```diff
@@ -0,0 +1,444 @@
+# Contributing to CG
+
+:+1::tada: First off, thanks for taking the time to contribute! :tada::+1:
+
+This is a guide for contributing to the CG package. Please check here first if you want to [set up an environment and develop](#local-development), [open and issue](#reporting-bugs), [suggest an enhancement](#suggesting-enhancements), [open a pull request](#pull-requests) etc.
+
+#### Table Of Contents
+
+[Code of Conduct](#code-of-conduct)
+
+[Branch Model](#branch-model)
+
+[How Can I Contribute?](#how-can-i-contribute)
+  * [Reporting Bugs](#reporting-bugs)
+  * [Suggesting Enhancements](#suggesting-enhancements)
+  * [Your First Code Contribution](#your-first-code-contribution)
+  * [Pull Requests](#pull-requests)
+
+[Styleguides](#styleguides)
+  * [Git Commit Messages](#git-commit-messages)
+  * [Python style guide](#python-styleguide)
+  * [Documentation Styleguide](#documentation-styleguide)
+
+[CG Design decisions](#design-decisions)
+  * [apps](#apps)
+  * [CLI](#cli)
+  * [meta](#meta)
+  * [server](#sever)
+  * [store](#store)
+
+
+
+
+## Code of Conduct
+
+Communicating around code can be a sensitive thing so please do your best to keep a positive tone. Remember that people are putting significant amount of work behind a PR or a review, stay humble :star:.
+
+
+## Branch model
+
+Cg is using github flow branching model as described in our [development manual][development-branch-model].
+
+
+## How Can I Contribute?
+
+### Reporting Bugs
+
+This section guides you through submitting a bug report for CG. Following these guidelines helps other developers and contributors understand your report :pencil:, reproduce the behavior :computer: :computer:, and find related reports :mag_right:.
+
+Before creating bug reports, please try to search the issues (opened and closed) if the problem has been described before, as you might find out that you don't need to create one. When you are creating a bug report, please [include as many details as possible](#how-do-i-submit-a-good-bug-report).
+
+> **Note:** If you find a **Closed** issue that seems like it is the same thing that you're experiencing, open a new issue and include a link to the original issue in the body of your new one.
```

Comment:
So one shouldn't reopen a closed and resolved reoccurring issue?

Reply:
I like this way of doing it. A closed issue can include lot's of stuff that is not relevant for the new situation